buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/PrecommitTasks.groovy

```diff
@@ -144,18 +146,23 @@ class PrecommitTasks {
         project.tasks.withType(CheckForbiddenApis).configureEach {
             dependsOn(buildResources)
 
-            //  use the runtime classpath if we have it, but some qa projects don't have one...
-            if (name.endsWith('Test')) {
-                FileCollection runtime = project.sourceSets.test.runtimeClasspath
-                if (runtime != null) {
-                    classpath = runtime.plus(project.sourceSets.test.compileClasspath)
-                }
-            } else {
-                FileCollection runtime = project.sourceSets.main.runtimeClasspath
-                if (runtime != null) {
-                    classpath = runtime.plus(project.sourceSets.main.compileClasspath)
+            String[] parts = name.split(ForbiddenApisPlugin.FORBIDDEN_APIS_TASK_NAME)
+            String sourceSetName = 'main'
+            if (parts.length == 2) {
+                sourceSetName = parts[1].toLowerCase(Locale.ROOT)
+            }
+            //add the sourceSet's compile classPath if it exists
+            SourceSetContainer sourceSets = project.sourceSets
+            if (sourceSetName) {
```

Comment:
Why do we need this leniency? If a CheckForbiddenApis task exists, this source set should exist shouldn't it? With this leniency, or that below, we could have a bug here in the name munging and silently skip over all of this?

Reply:
@rjernst good call. the leniency is not needed. Fixing that revealed a bug around casing which has also been fixed in https://github.com/elastic/elasticsearch/pull/54731/commits/cdf8253325727d62a227c415a49d041cdb681861